### PR TITLE
Player icons; login notification and activity indicator handling

### PR DESCRIPTION
**Pull request by mitchdowney**

### CODE REVIEW

**Comment by mitchdowney** [comment]
@kreonjr the Login table cell in the MoreTableVC still needs a constraint for the programmatic activity indicator...not sure how to do that at the moment.

File: Podverse/PVAuth.swift [hunk 1]
```diff
@@ -170,26 +173,36 @@ class PVAuth: NSObject {
         }
     }
     
+    fileprivate func notifyLoggingIn() {
+        DispatchQueue.main.async {
```
Comment (kreonjr) [inline Podverse/PVAuth.swift:177]: If you are dispatching the notification on the main queue here, is it necessary to dispatch the observers to the main queue as well? I noticed a lot of main dispatches and I think if you do it here, thats good enough

File: Podverse/PVAuth.swift [hunk 2]
```diff
@@ -170,26 +173,36 @@ class PVAuth: NSObject {
...
+            NotificationCenter.default.post(name:NSNotification.Name(rawValue: kLoggingIn), object: self, userInfo: nil)
+        }
+    }
+    
+    fileprivate func notifyLoginFailed() {
+        DispatchQueue.main.async {
```
Comment (kreonjr) [inline Podverse/PVAuth.swift:183]: same here. if the dispatch is on the main queue, the listeners shouldn't also need dispatching

**Comment by mitchdowney** [comment]
Derp, pretty sure I did that last commit wrong.

I don't understand. If we're keeping these two notifications on the main queue, which dispatches do we remove?

File: Podverse/MoreTableViewController.swift [hunk 3]
```diff
@@ -111,14 +155,58 @@ extension MoreTableViewController:UITableViewDelegate, UITableViewDataSource {
...
         }
     }
+    
+    @objc func loginFailed(_ notification:Notification) {
+        let indexPath = IndexPath(row: 1, section: 0)
+        guard let cell = self.tableView.cellForRow(at: indexPath) else {
+            return
+        }
+        
+        DispatchQueue.main.async {
```
Comment (kreonjr) [inline Podverse/MoreTableViewController.swift]: This can be removed.

File: Podverse/MoreTableViewController.swift [hunk 4]
```diff
@@ -111,14 +155,58 @@ extension MoreTableViewController:UITableViewDelegate, UITableViewDataSource {
...
 
 extension MoreTableViewController {
+    
+    @objc func loggingIn(_ notification:Notification) {
+        let indexPath = IndexPath(row: 1, section: 0)
+        guard let cell = self.tableView.cellForRow(at: indexPath) else {
+            return
+        }
+        
+        DispatchQueue.main.async {
```
Comment (kreonjr) [inline Podverse/MoreTableViewController.swift]: This can be removed.

File: Podverse/LoginViewController.swift [hunk 5]
```diff
@@ -23,9 +30,62 @@ class LoginViewController: PVViewController {
...
+
+    @objc func loggedInSuccessfully(_ notification:Notification) {
+        DispatchQueue.main.async {
+            self.activityIndicator.stopAnimating()
+            self.dismiss(animated: true, completion: nil)
+        }
     }
     
+    @objc func loginFailed(_ notification:Notification) {
+        DispatchQueue.main.async {
```
Comment (kreonjr) [inline Podverse/LoginViewController.swift]: This can be removed.

File: Podverse/LoginViewController.swift [hunk 6]
```diff
@@ -23,9 +30,62 @@ class LoginViewController: PVViewController {
...
+            self.noLoginNeededToText.isHidden = true
+            self.loginButton.isHidden = true
+            self.noThanksButton.isHidden = true
+            self.loggingInLabel.isHidden = false
+            self.activityIndicator.startAnimating()
+        }
+    }
+
+    @objc func loggedInSuccessfully(_ notification:Notification) {
+        DispatchQueue.main.async {
```
Comment (kreonjr) [inline Podverse/LoginViewController.swift]: This can be removed.

File: Podverse/LoginViewController.swift [hunk 7]
```diff
@@ -23,9 +30,62 @@ class LoginViewController: PVViewController {
...
+    
+    deinit {
+        removeObservers()
+    }
+    
+}
 
+extension LoginViewController {
+    @objc func loggingIn(_ notification:Notification) {
+        DispatchQueue.main.async {
```
Comment (kreonjr) [inline Podverse/LoginViewController.swift]: This can be removed.

File: Podverse/MoreTableViewController.swift [hunk 8]
```diff
@@ -111,14 +155,58 @@ extension MoreTableViewController:UITableViewDelegate, UITableViewDataSource {
...
+                if let activityIndicator = view as? UIActivityIndicatorView {
+                    activityIndicator.removeFromSuperview()
+                }
+            }
+            cell.textLabel?.text = "Login"
+        }
+    }
+    
     @objc func loggedOutSuccessfully(_ notification:Notification) {
         DispatchQueue.main.async {
```
Comment (kreonjr) [inline Podverse/MoreTableViewController.swift]: This can probably be removed

File: Podverse/PVAuth.swift [hunk 9]
```diff
@@ -174,15 +174,11 @@ class PVAuth: NSObject {
     }
     
     fileprivate func notifyLoggingIn() {
-        DispatchQueue.main.async {
```
Comment (kreonjr) [inline Podverse/PVAuth.swift]: Place this bad boy back

File: Podverse/PVAuth.swift [hunk 10]
```diff
@@ -174,15 +174,11 @@ class PVAuth: NSObject {
...
-            NotificationCenter.default.post(name:NSNotification.Name(rawValue: kLoggingIn), object: self, userInfo: nil)
-        }
+        NotificationCenter.default.post(name:NSNotification.Name(rawValue: kLoggingIn), object: self, userInfo: nil)
     }
     
     fileprivate func notifyLoginFailed() {
-        DispatchQueue.main.async {
```
Comment (kreonjr) [inline Podverse/PVAuth.swift]: And this guy too

**Pull request merged**